omnigan/trainer.py

```diff
@@ -1614,3 +1618,43 @@ def del_output_path(self, force=False):
 
         if (Path(self.opts.output_path) / "is_functional.test").exists() or force:
             shutil.rmtree(self.opts.output_path)
+
+    def compute_smog(self, x, z=None, d=None, s=None, use_sky_seg=False):
```

Comment:
I struggle a bit with understanding the role of variables with single-letter names (z, d, s, A, I...)

Reply:
I understand it can get a little confusing. All over the code, in general:

* `m` refers to "mask", either as input (`batch["data"]["m"]`) or output `G.decoders["m"]` or as a mask variable with shape `batch x 1 x height x width`
* `d` refers similarly to "depth"
* `s` to "segmentation" when it is a task or "simulated" when it is a domain
* `z` always refers to the latent vector, *i.e.* the output of the encoder, *i.e.* the input to the decoders
  * `z` is a tensor `batch x 2048 x 80 x 80` when the encoder is `deeplabv2`, it is a tuple (high_level_features, low_level_features) when the encoder is `deeplabv3`. High vs Low level features means we extract both low-level feature maps in the middle of the encoder and high-level feature maps as its output
  * in the case of dlv3 we may use the `mobilenet` or `resnet` backbones, one has fewer parameters than the other

In the special case of this function the other single letter vars are not standard and we should actually rename them

Reply:
updated @alexhernandezgarcia :)